sematic/ui/src/types/Types.tsx

```diff
@@ -541,14 +545,81 @@ function PlotlyFigureValueView(props: ValueViewProps) {
 
 function MatplotlibFigureValueView(props: ValueViewProps) {
   let { valueSummary } = props;
-  let { path } = valueSummary;
+
+  const hasFigureJsonData = useMemo(() => !!valueSummary['mpld3'], [valueSummary]);
+  
+  return hasFigureJsonData ? 
+    <MatplotlibFigureValueFigure key={valueSummary['mpld3']['id']} spec={valueSummary['mpld3']} />
+    : <MatplotlibFigureValueImage path={valueSummary['path']} /> ;
+}
+
+interface MatplotlibFigureValueImageProps{
+  path: string
+}
+
+function MatplotlibFigureValueImage(props: MatplotlibFigureValueImageProps) {
+  const { path } = props;
   return (
     <Zoom>
       <img src={path} width={"100%"} alt="matplotlib figure" />
     </Zoom>
   );
 }
 
+interface MatplotlibFigureValueFigureProps{
+  spec: any
+}
+
+function MatplotlibFigureValueFigure(props: MatplotlibFigureValueFigureProps) {
+  const { spec: specProp } = props;
+  const figureId = useMemo(() => `fig-${uuidv4()}`, []);
+
+  const drawFigure = useMatplotLib(figureId);
+
+  const [figDivRef, { width, height}] = useMeasure<HTMLDivElement>();
+
+  const hasFigureRendered = useRef(false)
+
+  const [scaleAndTranslate, setScaleAndTranslate] = useState({
+    scale: 1, translate: {x: 0, y: 0}
+  });
+
+  const [scaledHeight, setScaleHeight] = useState(0);
+
+  useLayoutEffect(() => {
+    // wait for the element's final layout when its width is fully expanded.
+    if (!hasFigureRendered.current && width > 0) {
```

Comment:
How about rendering the loading wheel we use for logs until the image is retrieved and rendered.
This would improve the experience for large images.

Reply:
Thanks for the suggestion. After some investigation, I found out that the artifacts (together with the chart JSON representation) had already been loaded earlier up in some parent component (from the `/runs` API). So there is no loading period when we render this component. 

In other words, a full-page loading indicator is already done somewhere at the parent level. 

If you are concerned about the `& width > 0` part, it happens in a couple of React rendering cycles, which happens almost instantly. There is no need to add a loading indicator in between, which could cause some flashing effects to damage the user experience.